src/uu/dd/src/dd.rs

```diff
@@ -832,10 +832,14 @@ fn gen_prog_updater(rx: mpsc::Receiver<ProgUpdate>, print_level: Option<StatusLe
             }
         });
 
+        let mut progress_as_secs = 0;
         while let Ok(update) = rx.recv() {
             // (Re)print status line if progress is requested.
-            if Some(StatusLevel::Progress) == print_level {
+            if Some(StatusLevel::Progress) == print_level
+                && update.duration.as_secs() == progress_as_secs
+            {
                 reprint_prog_line(&update);
+                progress_as_secs += 1;
```

Comment:
This breaks if the `ibs`, `obs` or `bs` are so large that the it takes longer than a second to copy because then `update.duration.as_secs()` is larger than `progress_as_secs`. The `progress_as_secs` variable then never gets updated, while the duration continues to increase. So we should check whether it's larger than or equal to the `progress_as_secs`
```suggestion
            if Some(StatusLevel::Progress) == print_level
                && update.duration.as_secs() >= progress_as_secs
            {
                reprint_prog_line(&update);
                progress_as_secs += update.duration.as_secs() + 1;
```
On my machine I tested this with `dd status=progress bs=1G < file1.txt > file2.txt` where `file1.txt` was an empty file truncated to a couple of gigabytes.

Reply:
I see what you mean, and I've updated it so it's similar to the suggested changes!

However, I realize even with these changes, `status=progress` won't rewrite once/sec because progress information isn't transmitted every second - it's only transmitted _after_ a reading and writing increment has completed (which if byte size is long enough, may take over a second).

Let me know how I should proceed with this issue, or if you have any suggestions on where I should take this - my current understanding of how we may want to fix this is to update/create a thread that polls progress instead of waiting for progress to be transmitted, but that might require more extensive changes than I initially anticipated, which may take a bit for me.

Reply:
Thank you!

> However, I realize even with these changes, status=progress won't rewrite once/sec because progress information isn't transmitted every second - it's only transmitted after a reading and writing increment has completed (which if byte size is long enough, may take over a second).

I think this happens with GNU too. Putting it in a separate thread is an interesting idea, but probably better suited to another PR.

Reply:
I'll go ahead and create another issue - for me, it seems like GNU's dd consistently polls every second, but let me know if there's anything else I can do for this PR!

Reply:
> it seems like GNU's dd consistently polls every second

Interesting, I wonder what the difference is. On my machine it sometimes works correctly, but then suddenly hangs for a long time. For example, here is the final output of a command copying 4G:
```
❯ dd status=progress bs=1G < 1 > 2                                                                                                                                00:13:33
4294967296 bytes (4,3 GB, 4,0 GiB) copied, 3 s, 1,4 GB/s
4+0 records in
4+0 records out
4294967296 bytes (4,3 GB, 4,0 GiB) copied, 11,7511 s, 365 MB/s
```
It worked for 3 seconds and then hanged for 8 more seconds. Then when I tried again, it hanged for the first 8 seconds, showing no output at all and then worked for the last few seconds. Anyway, it's not something that we have to be concerned about. If we can do better somehow, that'd be great!